yarn-project/aztec.js/src/contract/unsafe_contract.ts

```diff
@@ -0,0 +1,19 @@
+import { ContractArtifact } from '@aztec/foundation/abi';
+import { ContractInstanceWithAddress } from '@aztec/types/contracts';
+
+import { Wallet } from '../wallet/index.js';
+import { ContractBase } from './contract_base.js';
+
+/** Unsafe constructor for ContractBase that bypasses the check that the instance is registered in the wallet. */
```

Comment:
I don't understand how/where any bypass takes place since we're just calling `super`?

Reply:
The constructor in the base class is protected, this class makes it public.